test/Microsoft.AspNetCore.Server.KestrelTests/PipelineExtensionTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.KestrelTests
+{
+    public class PipelineExtensionTests : IDisposable
+    {
+        // ulong.MaxValue.ToString().Length
+        private const int _ulongMaxValueLength = 20; 
+
+        private readonly IPipe _pipe;
```

Comment:
Ok. I'm still sad but I can accept this. I think.

I like tests to be self-contained. They can call helper methods and such, but generally I avoid having class state like the plague. Ideally every test should be a self-contained mini spec of an expected behavior of the system.